Extensions/NavMeshPathfindingBehavior/navmeshpathfindingobstacleruntimebehavior.ts

```diff
@@ -0,0 +1,368 @@
+        runtimeScene
+      );
+      manager.setAreaBounds(left, top, right, bottom);
+    }
+
+    public setAreaBounds(left: float, top: float, right: float, bottom: float) {
+      this._areaLeftBound = left;
+      this._areaTopBound = top;
+      this._areaRightBound = right;
+      this._areaBottomBound = bottom;
+      this.invalidateNavMesh();
+    }
+
+    public static setCellSize(runtimeScene: any, cellSize: float) {
+      const manager = NavMeshPathfindingObstaclesManager.getManager(
+        runtimeScene
+      );
+      manager.setCellSize(cellSize);
+    }
+
+    public setCellSize(cellSize: float) {
+      this._cellSize = cellSize;
+      this.invalidateNavMesh();
+    }
+
+    public static getAreaLeftBound(runtimeScene: any): float {
+      const manager = NavMeshPathfindingObstaclesManager.getManager(
+        runtimeScene
+      );
+      return manager.getAreaLeftBound();
+    }
+
+    public static getAreaToptBound(runtimeScene: any): float {
+      const manager = NavMeshPathfindingObstaclesManager.getManager(
+        runtimeScene
+      );
+      return manager.getAreaTopBound();
+    }
+
+    public static getAreaRightBound(runtimeScene: any): float {
+      const manager = NavMeshPathfindingObstaclesManager.getManager(
+        runtimeScene
+      );
+      return manager.getAreaRightBound();
+    }
+
+    public static getAreaBottomBound(runtimeScene: any): float {
+      const manager = NavMeshPathfindingObstaclesManager.getManager(
+        runtimeScene
+      );
+      return manager.getAreaBottomBound();
+    }
+
+    public static getCellSize(runtimeScene: any): float {
+      const manager = NavMeshPathfindingObstaclesManager.getManager(
+        runtimeScene
+      );
+      return manager.getCellSize();
+    }
+
+    public getAreaLeftBound(): float {
+      return this._areaLeftBound;
+    }
+
+    public getAreaTopBound(): float {
+      return this._areaTopBound;
+    }
+
+    public getAreaRightBound(): float {
+      return this._areaRightBound;
+    }
+
+    public getAreaBottomBound(): float {
+      return this._areaBottomBound;
+    }
+
+    public getCellSize(): float {
+      return this._cellSize;
+    }
+
+    getNavMesh(obstacleCellPadding: integer): NavMesh {
+      let navMesh = this._navMeshes.get(obstacleCellPadding);
+      if (!navMesh) {
+        const grid = new gdjs.RasterizationGrid(
+          this._areaLeftBound,
+          this._areaTopBound,
+          this._areaRightBound,
+          this._areaBottomBound,
+          this._cellSize,
+          // make cells square in the world
+          this._cellSize / this._isometricRatio
+        );
+        gdjs.ObstacleRasterizer.rasterizeObstacles(grid, this._obstacles);
+        gdjs.RegionGenerator.generateDistanceField(grid);
+        gdjs.RegionGenerator.generateRegions(grid, obstacleCellPadding);
+        //TODO make threshold configurable ?
+        // 1 seems to be a good value.
+        const threshold = 1;
```

Comment:
It is the maximum distance (in cells) the edge of the contour may deviate from the source geometry when the rastered obstacles are vectorized.

Setting it to:
- 1 ensure that an aliased edge won't be split to more edges.
- more that 1 will reduce the number of edges but the obstacles edges will be followed with less accuracy.
- less than 1 might be more accurate but it may try to follow the aliasing and be a lot less accurate.

For instance, with threshold = 0.1, the aliasing is followed and it creates a lot of polygons.
![Iso0 1](https://user-images.githubusercontent.com/2611977/129759755-8ee92fc5-b4c5-4272-9bef-f2adb5217654.png)
Note that a lower threshold won't make up for corners that are forgotten at the rasterization step.

Whereas at 0.71 or 1 the edges are better rebuilt:
![Iso0 71](https://user-images.githubusercontent.com/2611977/129759775-bc95f374-db2b-42b3-8b9c-73ef9f2c22c8.png)

Values under 1 can be useful in specific cases:
- when edges are horizontal or vertical, there is no aliasing so value near 0 can do better results.
- when edges are 45° multiples, aliased vertex won't be farther than sqrt(2)/2 so values over 0.71 should give good results but not necessarily better than 1.

For instance, with a threshold of 0.1:
![Squares0 1](https://user-images.githubusercontent.com/2611977/129759804-03a72e07-c40b-4491-840b-0435f939f3cf.png)

But greater values (here, 1) are a good option to reduce the number of vertices:
![Squares1](https://user-images.githubusercontent.com/2611977/129759830-d3f74fe2-b49a-4776-ab12-ff87256144cd.png)